InteractiveHtmlBom/ecad/genericjson.py

```diff
@@ -65,6 +67,48 @@ def _verify(self, pcb):
 
         return True
 
+    def _texts(self):
+        for layer in self.pcbdata['drawings'].values():
+            for side in layer.values():
+                for dwg in side:
+                    if 'text' in dwg:
+                        yield dwg
+
+    @staticmethod
+    def _remove_control_codes(s):
+        import unicodedata
+        return ''.join(c for c in s if unicodedata.category(c)[0] != "C")
+
+    def _parse_font_data(self):
+        font_parser = FontParser()
+        for dwg in self._texts():
+            if 'svgpath' not in dwg:
+                dwg['text'] = self._remove_control_codes(dwg['text'])
+                font_parser.parse_font_for_string(dwg['text'])
+
+        if font_parser.get_parsed_font().items():
+            self.pcbdata['font_data'] = {}
+            for (c, wl) in font_parser.get_parsed_font().items():
+                if c not in self.pcbdata['font_data']:
+                    self.pcbdata['font_data'][c] = wl
```

Comment:
You can assign get_parsed_font() result directly it's a dict already, no need to deduplicate.